starfyre/__main__.py

```diff
@@ -41,20 +41,38 @@ def create_main_file(path):
 
 
 @click.command()
-@click.option("--path", default=".", help="Path to the project")
-@click.option("--dev", default=False, help="Start the compilation and generate the build package.")
-@click.option("--build", default=False, help="Start the build package")
-def main(path, dev, build):
-    if dev:
-        path_ = path + "/__init__.py"
-        # get absolute path
-        path = os.path.abspath(path_)
-        compile(path)
-        create_main_file(os.path.dirname(path))
+@click.option("--path", default=".", help="Path to the project directory")
+@click.option("--build", is_flag=True, help="Compile and build package")
+def main(path, build):
+    """
+    Command-line interface to compile and build a Starfyre project.
+
+    Args:
+
+        path (str): Path to the project directory.\n
+        build (bool): Whether to start the build package.
+    """
+
+    # Convert path to absolute path
+    absolute_path = Path(path).resolve()
+
+    if not absolute_path.exists() and build:
+        click.echo(
+            "Error: Please provide a valid path using the --path flag.\nUse --help for more details")
+        return
+
+
     if build:
+        # Compile and build project
+        build_dir = absolute_path / "build"
+        init_file_path = absolute_path / "__init__.py"
+        compile(init_file_path.resolve())
+        create_main_file(str(absolute_path))
+
+        # Start/run project
         subprocess.run(
             [sys.executable, "-m", "build"],
-            cwd=path,
+            cwd=build_dir.resolve(),
```

Comment:
@DeleMike , this is the issue here. This needs to be path and not `build_dir`

Reply:
Thanks, let me look into it more!